services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCIssuerEndpoint.java

```diff
@@ -301,39 +306,90 @@ public Response requestCredential(
+            }
+            // Then for format. We know spec does not allow both parameter. But we are tolerant if you send both
+            // Was found by id, check that the format matches.
+            if (requestedFormat!=null && !requestedFormat.equals(supportedCredentialConfiguration.getFormat())){
+                LOGGER.debugf("Credential with configuration id %s does not support requested format %s, but supports %s.", requestedCredentialId, requestedFormat, supportedCredentialConfiguration.getFormat());
+                throw new BadRequestException(getErrorResponse(ErrorType.UNSUPPORTED_CREDENTIAL_FORMAT));
+            }
+        }
+
+        if(supportedCredentialConfiguration==null && requestedFormat!=null) {
+            // Search by format
+            supportedCredentialConfiguration = getSupportedCredentialConfiguration(credentialRequestVO, supportedCredentials, requestedFormat);
+            if(supportedCredentialConfiguration==null) {
+                LOGGER.debugf("Credential with requested format %s, not supported.", requestedFormat);
+                throw new BadRequestException(getErrorResponse(ErrorType.UNSUPPORTED_CREDENTIAL_FORMAT));
+            }
         }
 
         CredentialResponse responseVO = new CredentialResponse();
 
-        Object theCredential = getCredential(userSessionModel, supportedCredentialConfiguration.getScope(), credentialRequestVO.getFormat());
-        switch (requestedFormat) {
-            case LDP_VC, JWT_VC, SD_JWT_VC -> responseVO.setCredential(theCredential);
-            default -> throw new BadRequestException(
-                    getErrorResponse(ErrorType.UNSUPPORTED_CREDENTIAL_TYPE));
+        Object theCredential = getCredential(authResult, supportedCredentialConfiguration, credentialRequestVO);
+        if(Objects.equals(requestedFormat, Format.JWT_VC) || Objects.equals(requestedFormat, LDP_VC) || Objects.equals(requestedFormat, SD_JWT_VC)) {
+            responseVO.setCredential(theCredential);
+        } else {
+            throw new BadRequestException(getErrorResponse(ErrorType.UNSUPPORTED_CREDENTIAL_TYPE));
         }
-        return Response.ok().entity(responseVO)
-                .build();
+        return Response.ok().entity(responseVO).build();
+    }
+
+    private SupportedCredentialConfiguration getSupportedCredentialConfiguration(CredentialRequest credentialRequestVO, Map<String, SupportedCredentialConfiguration> supportedCredentials, String requestedFormat) {
+        // 1. Format resolver
+        List<SupportedCredentialConfiguration> configs = supportedCredentials.values().stream()
+                .filter(supportedCredential -> Objects.equals(supportedCredential.getFormat(),requestedFormat))
+                .collect(Collectors.toList());
+
+        List<SupportedCredentialConfiguration> matchingConfigs;
+
+        if(Objects.equals(requestedFormat, Format.JWT_VC) || Objects.equals(requestedFormat, LDP_VC)) {
```

Comment:
> How about importing JWT_VC the same as LDP_VC and SD_JWT_VC?
> import static org.keycloak.protocol.oid4vc.model.Format.JWT_VC;

Thanks

> I am not 100% sure but it might be replaced as follows:
>   List.of(SD_JWT_VC, LDP_VC, JWT_VC).contains(requestedFormat)

Wouldn't get the purpose. As i am resolving case per case.

> I think that SD_JWT_VC is more plausible than other two VC format, to how about the following?
> if (Objects.equals(requestedFormat, SD_JWT_VC) || Objects.equals(requestedFormat, LDP_VC) || Objects.equals(requestedFormat, JWT_VC)) {

I don't need to check all of them in one shot, but will start with JWT_VC

Reply:
Resolved with [5f99e16](https://github.com/keycloak/keycloak/pull/30692/commits/5f99e1671d30c85c0718a56dd88d3508f4a20d09)